src/main/java/uk/gov/hmcts/reform/sandl/snlapi/config/BeanConfiguration.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.sandl.snlapi.config;
+
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.web.client.DefaultResponseErrorHandler;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.sandl.snlapi.exceptions.OptimisticLockException;
+
+import java.io.IOException;
+
+@Configuration
+public class BeanConfiguration {
+
+    @Bean
+    public RestTemplate restTemplate(RestTemplateBuilder builder) {
+        return builder.errorHandler(
+            new DefaultResponseErrorHandler() {
+                @Override
+                public void handleError(ClientHttpResponse response) throws IOException {
+                    if (response.getStatusCode() == HttpStatus.CONFLICT) {
+                        throw new OptimisticLockException();
```

Comment:
I think we should somehow test this behaviour but I'd propose to do it in integration tests later on